QuEST/src/CPU/QuEST_cpu.c

```diff
@@ -942,6 +942,40 @@ qreal densmatr_calcHilbertSchmidtDistanceSquaredLocal(Qureg a, Qureg b) {
     return trace;
 }
 
+/** computes Tr(conjTrans(a) b) = sum of (a^* b) */
+qreal densmatr_calcHilbertSchmidtScalarProductLocal(Qureg a, Qureg b) {
+    
+    long long int index;
+    long long int numAmps = a.numAmpsPerChunk;
+        
+    qreal *aRe = a.stateVec.real;
+    qreal *aIm = a.stateVec.imag;
+    qreal *bRe = b.stateVec.real;
+    qreal *bIm = b.stateVec.imag;
+    
+    qreal trace = 0, traceIm = 0;
+    
+# ifdef _OPENMP
+# pragma omp parallel \
+    shared    (aRe,aIm, bRe,bIm, numAmps) \
+    private   (index, traceIm) \
```

Comment:
This looks bugged; `traceIm` can't be OpenMP `private` yet be accumulated in the loop. It seems `traceIm` was also meant to be in `reduction`

Reply:
removed this in bb79977ff0cddaf55527bb0dc1221e993bd466ef